packages/flutter/test/foundation/tree_diagnostics_mixin_test.dart

```diff
@@ -2,64 +2,1395 @@
+      '  child node A: TestTree#00000\n'
+      '  child node B: TestTree#00000:\n'
+      '    child node B1: TestTree#00000\n'
+      '    child node B2: TestTree#00000\n'
+      '    child node B3: TestTree#00000\n'
+      '  child node C: TestTree#00000\n',
+    );
+
+    // Single line mode does not display children.
+    goldenStyleTest(
+      'single line',
+      style: DiagnosticsTreeStyle.singleLine,
+      golden: 'TestTree#00000');
+  });
+
+  test('TreeDiagnosticsMixin tree with properties test', () async {
+    void goldenStyleTest(String description, {
+      DiagnosticsTreeStyle style,
+      DiagnosticsTreeStyle lastChildStyle,
+      @required String golden,
+    }) {
+      final TestTree tree = new TestTree(
+        properties: <DiagnosticsNode>[
+          new StringProperty('stringProperty1', 'value1'),
+          new DoubleProperty('doubleProperty1', 42.5),
+          new DoubleProperty(
+            'roundedProperty',
+            1.0 / 3.0,
+          ),
+          new StringProperty(
+            'DO_NOT_SHOW',
+            'DO_NOT_SHOW',
+            hidden: true,
+          ),
+          new DiagnosticsProperty<Object>(
+            'DO_NOT_SHOW_NULL',
+            null,
+            defaultValue: null,
+          ),
+          new DiagnosticsProperty<Object>(
+            'nullProperty',
+            null,
+          ),
+          new StringProperty(
+            'node_type',
+            '<root node>',
+            showName: false,
+          ),
+        ],
+        children: <TestTree>[
+          new TestTree(name: 'node A', style: style),
+          new TestTree(
+            name: 'node B',
+            properties: <DiagnosticsNode>[
+              new StringProperty('p1', 'v1'),
+              new StringProperty('p2', 'v2'),
+            ],
+            children: <TestTree>[
+              new TestTree(name: 'node B1', style: style),
+              new TestTree(
+                name: 'node B2',
+                properties: <DiagnosticsNode>[
+                  new StringProperty('property1', 'value1'),
+                ],
+                style: style,
+              ),
+              new TestTree(
+                name: 'node B3',
+                properties: <DiagnosticsNode>[
+                  new StringProperty(
+                    'node_type',
+                    '<leaf node>',
+                    showName: false),
+                  new IntProperty('foo', 42),
+                ],
+                style: lastChildStyle ?? style,
+              ),
+            ],
+            style: style,
+          ),
+          new TestTree(
+            name: 'node C',
+            properties: <DiagnosticsNode>[
+              new StringProperty('foo', 'multi\nline\nvalue!'),
+            ],
+            style: lastChildStyle ?? style,
+          ),
+        ],
+        style: lastChildStyle,
+      );
+
+      // Change the expectation line when the tree generation code in this test is fixed.
```

Comment:
Actually... removed comment as the expectations were already changed due to tree generation being fixed.